### PR TITLE
Make segment size configurable and change default to 1GB

### PR DESCRIPTION
**Pull request by DeepThinker666**
Make segment size configurable and change default to 1GB, to #1550

### CODE REVIEW

File: be/src/storage/rowset/rowset_writer.h [hunk 1]
```diff
@@ -124,6 +124,7 @@ class RowsetWriter {
 
 protected:
     vectorized::DictColumnsValidMap _global_dict_columns_valid_info;
+    const uint64_t MAX_SEGMENT_SIZE = static_cast<uint64_t>(config::max_segment_file_size * config::segment_file_scale);
```
Comment (imay) [inline be/src/storage/rowset/rowset_writer.h:127]: I have question, why does it need two configuration to determinate segment size?
Why not use only one configuration?

Comment (DeepThinker666) [reply be/src/storage/rowset/rowset_writer.h:127]: I also have no idea why there is a ratio, maybe for the inaccuracy of segment size calculation. I just keep the original logical.

Comment (DeepThinker666) [reply be/src/storage/rowset/rowset_writer.h:127]: I also agree to just use only one configuration. I will change it.

**Review by imay — DISMISSED** [review]
LGTM

File: be/src/storage/rowset/rowset_writer.h [hunk 2]
```diff
@@ -124,6 +124,7 @@ class RowsetWriter {
 
 protected:
     vectorized::DictColumnsValidMap _global_dict_columns_valid_info;
+    const uint64_t MAX_SEGMENT_SIZE = config::max_segment_file_size;
```
Comment (sduzh) [inline be/src/storage/rowset/rowset_writer.h:127]:
```suggestion
    const uint64_t _max_segment_size = config::max_segment_file_size;
```
This is a normal class member variable.

Comment (sduzh) [reply be/src/storage/rowset/rowset_writer.h:127]: BTW, why not just using `config::max_segment_file_size` directly?

Comment (DeepThinker666) [reply be/src/storage/rowset/rowset_writer.h]: I use config::max_segment_file_size to replace MAX_SEGMENT_SIZE directly

**Review by sduzh — DISMISSED** [review]

**Comment by DeepThinker666** [comment]
run starrocks_be_unittest

**Comment by DeepThinker666** [comment]
run starrocks_be_unittest

File: be/src/common/config.h [hunk 3]
```diff
@@ -653,6 +653,8 @@ CONF_Bool(enable_schema_change_vectorized, "true");
 // max hdfs file handle
 CONF_mInt32(max_hdfs_file_handle, "1000");
 
+CONF_Int64(max_segment_file_size, "1073741824");
```
Comment (chaoyli) [inline be/src/common/config.h:656]: 1. segment_file_size is a better name. 
2. when to set the segment_file_size?

**Review by sduzh — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**